amr-wind/core/vs/tensor.H

```diff
@@ -170,16 +170,19 @@ struct TensorT
         return vv[pos];
     }
 
-    AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE T* data() noexcept { return vv; }
+    AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE T* data() noexcept
+    {
+        return vv.data();
+    }
     AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE const T* data() const noexcept
     {
-        return vv;
+        return vv.data();
     }
 
-    iterator begin() noexcept { return vv; }
-    iterator end() noexcept { return vv + ncomp; }
-    const_iterator cbegin() const noexcept { return vv; }
-    const_iterator cend() const noexcept { return vv + ncomp; }
+    iterator begin() noexcept { return vv.begin(); }
+    iterator end() noexcept { return vv.end(); }
+    const_iterator cbegin() const noexcept { return vv.cbegin(); }
+    const_iterator cend() const noexcept { return vv.cend(); }
```

Comment:
Are these consistent?
Do "cbegin" and "cend" exist?
Is there a reason not to use just "begin" and "end"?

Reply:
I read about these some more and it looks like this is all good. They will just return the correct type so it is good.